packages/pipeline-services/src/migration/migrateV1/index.ts

```diff
@@ -14,44 +14,58 @@
  * limitations under the License.
  */
 
-function migrate(pipeline: any) {
-  if (pipeline.pipelines[0].app_data) {
-    // title -> name
-    pipeline.pipelines[0].app_data.name = pipeline.pipelines[0].app_data.title;
-    delete pipeline.pipelines[0].app_data.title;
+const hasAppdataField = (node: any, fieldName: string): boolean => {
+  return (
+    Object.prototype.hasOwnProperty.call(node, "app_data") &&
+    Object.prototype.hasOwnProperty.call(node["app_data"], fieldName)
```

Comment:
why?

Reply:
there are issues with checking existence of falsy values, so I decided to just use the previous implementation using `Object.prototype.hasOwnProperty`

Reply:
@ajbozarth Does this work?
```ts
!!node.app_data?.hasOwnProperty(fieldName)
```

Reply:
I can check, I didn't try that one, after a hour-is of debugging I gave up and just used the code we had previously since I knew it worked

Reply:
That does work, just pushed the update